CONTRIBUTING.md

```diff
@@ -4,22 +4,74 @@ First off, thanks for taking the time to contribute!
 
 The following is a set of guidelines and helpful information for contributing to the 3D Artist's Handbook, which is hosted on [GitHub](https://github.com/sidney-eliot/3d-artists-handbook). These are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.
 
-#### Table Of Contents
-
-ToDo
+## Table of Contents
+
+- [Code of Conduct](#code-of-conduct)
+- [What Should I Know Before I Get Started?](#what-should-i-know-before-i-get-started)
+   - [Prerequisites](#prerequisites)
+   - [Choosing What to Work On](#choosing-what-to-work-on)
+- [How Can I Contribute?](#how-can-i-contribute)
+   - [Test Locally Before You Push](#test-locally-before-you-push)
+- [Styleguides](#styleguides)
+   - [Handbook Pages Styleguide](#handbook-pages-styleguide)
+      - [General](#general)
+      - [Specific](#specific)
+- [Additional Note](#additional-notes)
 
 ## Code of Conduct
+
 This project and everyone participating in it is governed by the [3D Artist's Handbook Code of Conduct](CODE_OF_CONDUCT.md). By participating, you are expected to uphold this code.
 
-## What should I know before I get started?
+## What Should I Know Before I Get Started?
 
-ToDo
+### Prerequisites
+
+1. You will need to have [Node.js](https://nodejs.org/en/download/package-manager), and thus `npm` available on your
+   terminal of choice.
+
+   > [!NOTE]
+   > Installing the latest version should work without much problem. However, just in case as a reminder, as of 
+   > 2024-12-10, Quartz 4 needs _at least_ Node v20 and up, as well as `npm` v9.3.1 and up to work.
+   >
+   > By the time you read this, the information contained here may be outdated, for up to date information on Quartz's
+   > dependencies, please consult Quartz's own documentation, which should be found [here](https://quartz.jzhao.xyz/).
+
+2. Then, clone this repository in a directory of your choosing.
+
+3. Change directories (`cd`) to be inside the repository you just cloned.
+
+4. Inside the repository, execute the following commands:
+
+   ```shell
+   npm i
+   npx quartz build --serve
+   ```
+
+5. Now, open a new browser window with the address: [https://localhost:8080](https://localhost:8080)  
+   Note: If the above commands worked properly, you should see a prompt with this address as a result.
+
+### Choosing What to Work On
+
+Before writing anything, please try to familiarize yourself with the [styleguides](#styleguides) below.
+
+Next, you're free to choose what area of the book you to contribute to. Please note however that areas marked with a 🚧
```

Comment:
> Maybe we should think about how exactly work in progress should be used and if it should be displayed on the public website or maybe only stay in the "work_in_progress" folder. I also think that stuff in the work in progress section that is viewable on the live build, should be in a state where it can be read so not too many factual mistakes and no spelling / grammar mistakes. For things that are still in their rougher stages, they should probably be kept in the "work_in_progress" folder until they are presentable

I agree with this, I think we can still publish stuff that is mostly done (grammar, facts and logic), if it is only missing content sections. In other words, stuff that is proofread at least once, but has some content headings/sections missing is okay as a would be a *coming seen* type of section.
The rest can be left in another branch, say a `dev` branch or maybe `alpha` / `beta` stages.

> Alternatively, maybe there is a way to add a section at the end of a page where work in progress stuff can be added that is not rendered on the website by quartz / node js.

This to seems a bit complicated, could you elaborate?
Personally, I'm thinking we could potentially comment these sections out.

> Also, don't be surprised when you find horrific spelling and completely broken sentences in the "work_in_progress" folder, the to-add.md file for example is just a long list of many little things that I quickly wrote down before I forget them of changes to make.

That's okay 😄. You make me think though, do you have a to-do system or list for this project, other than the to-add.md file?
Maybe we could use GitHub's task list/kanban board in the "Projects" tab?